dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/NettyChain.java

```diff
@@ -131,155 +135,155 @@ public synchronized void stop() {
             stop();
-            while (state.get() != ChainState.STOPPED) {
-                try {
-                    wait();
-                } catch (InterruptedException e) {
-                    Thread.currentThread().interrupt();
-                }
-            }
         }
     }
 
     @Override
     public synchronized void update(String resolvedHostName) {
 
-        updateCount++;
-
-        MSP.log("Updating NettyChain: " + endpointName + " Update count: " + updateCount + " Current state: " + state.get());
-
         if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
-            Tr.entry(this, tc, "update chain " + this);
+            Tr.entry(this, tc, "Updating Netty Chain  " + endpointName + " Current state: " + state.get());
         }
 
         if (!enabled || FrameworkState.isStopping()) {
-            MSP.log("Chain is disabled or framework is stopping, skipping update.");
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.exit(this, tc, "Chain is disabled or framework is stopping, skipping update ");
+            }
             return;
         }
 
         final ActiveConfiguration oldConfig = currentConfig;
-        final ActiveConfiguration newConfig;
-
-        // The old configuration was valid if it existed and was correctly configured
-        final boolean validOldConfig = (Objects.isNull(oldConfig)) ? false : oldConfig.validConfiguration;
-
-        newConfig = new ActiveConfiguration(isHttps(), getOwner().getTcpOptions(), isHttps() ? getOwner().getSslOptions() : null, getOwner().getHttpOptions(), getOwner().getRemoteIpConfig(), getOwner().getCompressionConfig(), getOwner().getSamesiteConfig(), getOwner().getHeadersConfig(), getOwner().getEndpointOptions(), resolvedHostName);
+        final ActiveConfiguration newConfig = new ActiveConfiguration(isHttps(), getOwner().getTcpOptions(), isHttps() ? getOwner().getSslOptions() : null, 
+                        getOwner().getHttpOptions(), getOwner().getRemoteIpConfig(), getOwner().getCompressionConfig(), 
+                        getOwner().getSamesiteConfig(), getOwner().getHeadersConfig(), getOwner().getEndpointOptions(), resolvedHostName);
+        
 
         if (newConfig.configPort < 0 || !newConfig.complete()) {
             if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
                 Tr.debug(this, tc, "Stopping chain due to configuration " + newConfig);
             }
-            MSP.log("Stopping chain due to configuration " + newConfig);
             // save the new/changed configuration before we start setting up the new chain
             currentConfig = newConfig;
             stopAndWait();
+            state.set(ChainState.UNINITIALIZED);
         }
 
-        if (!newConfig.unchanged(oldConfig)) {
-            MSP.log("This configuration differs and should cause an update");
+        if (!newConfig.unchanged(oldConfig) || state.get() != ChainState.STARTED) {
```

Comment:
This should be an `else if`, otherwise this could cause race conditions. Also `state.get() != ChainState.STARTED` is overkill think we just need the config.